app/helpers/configuration_helper.rb

```diff
@@ -1,3 +1,31 @@
 module ConfigurationHelper
   include_concern 'ConfigurationViewHelper'
+
+  # Model class name comparison method that sorts provider models first
+  def compare_decorator_class(a, b)
+    if a.to_s.starts_with?('ManageIQ::Providers') == b.to_s.starts_with?('ManageIQ::Providers')
+      a.to_s <=> b.to_s
+    elsif a.to_s.starts_with?('ManageIQ::Providers')
+      -1
+    elsif b.to_s.starts_with?('ManageIQ::Providers')
+      1
+    end
+  end
+
+  # Returns with a hash of allowed quadicons for the current user
+  def allowed_quadicons
+    MiqDecorator.descendants # Get all the decorator classes
+                .select { |klass| klass.new(nil).respond_to?(:quadicon) } # Select only the decorators that define a quadicon
```

Comment:
does `method_defined?(:quadicon)` work instead of `klass.new(nil).respond_to?(:quadicon)`?

Reply:
It does and I like it more, calling `new` 145 times less :wink: